src/meta/meta_bulk_load_service.cpp

```diff
@@ -1024,10 +1032,41 @@ void bulk_load_service::update_app_status_on_remote_storage_reply(const app_bulk
     }
 }
 
-// ThreadPool: THREAD_POOL_DEFAULT
+// ThreadPool: THREAD_POOL_META_STATE
 void bulk_load_service::partition_ingestion(const std::string &app_name, const gpid &pid)
 {
-    FAIL_POINT_INJECT_F("meta_bulk_load_partition_ingestion", [](dsn::string_view) {});
+    FAIL_POINT_INJECT_F("meta_bulk_load_partition_ingestion", [=](dsn::string_view) {
+        if (_apps_ingesting_count[pid.get_app_id()] < FLAGS_bulk_load_ingestion_concurrent_count) {
+            _apps_ingesting_count[pid.get_app_id()]++;
+        }
+    });
+
+    {
+        zauto_read_lock l(_lock);
+        if (_apps_ingesting_count[pid.get_app_id()] >= FLAGS_bulk_load_ingestion_concurrent_count) {
+            dwarn_f("app({}) has already {} partitions executing ingestion, partition({}) will "
+                    "wait and try it later",
+                    app_name,
+                    _apps_ingesting_count[pid.get_app_id()],
+                    pid);
+            tasking::enqueue(
+                LPC_META_STATE_NORMAL,
+                _meta_svc->tracker(),
+                std::bind(&bulk_load_service::partition_ingestion, this, app_name, pid),
+                pid.thread_hash(),
+                std::chrono::seconds(5));
+            return;
+        }
+    }
+
+    auto app_status = get_app_bulk_load_status(pid.get_app_id());
+    if (app_status != bulk_load_status::BLS_INGESTING) {
```

Comment:
? isn't = bulk_load_status::BLS_INGESTING, so that we ingnore it?

Reply:
Function `partition_ingestion` should only be called when app bulk load status is INGESTING. As we adds count restriction, and the delayed request will be resent after 5 seconds, app status may be changed.

Reply:
Oh, however, if one replica has been `INGESTING` status, shouldn't we ingnore it?

Reply:
Oh, I see, `get_app_bulk_load_status` is getting meta local status?

Reply:
If all partitions' status is INGESTING, app status will be INGESTING. `get_app_bulk_load_status` is to get app status in meta server locally.